sample_tracking/app/samples_table.tsx

```diff
@@ -160,19 +163,42 @@ export default function SamplesTable(props: SampleDataProps) {
 
     const onDeleteSampleClick = useCallback(
         (row: MRT_Row<Sample>) => {
-            if (!confirm(`Are you sure you want to delete ${row.code_lab}`)) {
-                return;
+            const deleteSampleFunction = () => {
+                let collectionName = `${row.trusted}_samples`;
+                const deletedDocRef = doc(db, collectionName, row.code_lab);
+                deleteDoc(deletedDocRef);
+                //send api delete request here, then refetch or update local table data for re-render
+                const updatedSamples = sampleData.samples.slice();
+                let index;
+                updatedSamples.forEach((sample: Sample, sampleIndex: number) => {
+                    if (sample.code_lab === row.code_lab) {
+                        index = sampleIndex;
+                        return;
+                    }
+                });
+                if (index) {
+                    updatedSamples.splice(index, 1);
+                    setSampleData({
+                        samples: updatedSamples,
+                        hasBeenUpdated: true,
+                    });
+                } else {
+                    console.log(`Error: Unable to find index for ${row.code_lab} and could not remove row from table.`);
```

Comment:
if this error shows up in the UI, it also needs to come from the localization table

Reply:
Resolving comment because this string does not show up in UI just the console. If we decide these strings should be translated I'll submit a followup PR translating all console messages.